soccer/Robot.hpp

```diff
@@ -187,7 +215,14 @@ class OurRobot: public Robot {
 	 * @brief Move to a given point using the default RRT planner
 	 * @param endSpeed - the speed we should be going when we reach the end of the path
 	 */
-	void move(const Geometry2d::Point &goal, float endSpeed = 0);
+	void move(const Geometry2d::Point &goal, Geometry2d::Point endVelocity = Geometry2d::Point());
```

Comment:
Fair point (ha, get it?) - we've been passing them by reference in most places, it hadn't crossed my mind that they're the same size as a pointer on 64 bit machines, so by-value makes more sense.

I've done some investigation and I'm with you that they should be by value.  Check out my benchmark and results here: https://gist.github.com/justbuchanan/a1594aead5fda4e6e526.  It's interesting that with no compiler optimizations by-reference is faster, but with them on, by-value is slightly faster.  Compilers are ridiculous.

I'd say let's leave it as is here, then once this PR is merged, I'll do a find and replace and make them all by value.